File: backend/src/main/java/com/example/service/PaymentService.java
package com.example.service;

import com.example.model.Payment;
import com.example.repository.PaymentRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.time.LocalDateTime;
import java.util.List;
import java.util.Optional;
import java.util.UUID;

@Service
public class PaymentService {
    
    @Autowired
    private PaymentRepository paymentRepository;
    
    public List<Payment> getAllPayments() {
        return paymentRepository.findAll();
    }
    
    public Optional<Payment> getPaymentById(Long id) {
        return paymentRepository.findById(id);
    }
    
    public Optional<Payment> getPaymentByOrderId(Long orderId) {
        return paymentRepository.findByOrderId(orderId);
    }
    
    public List<Payment> getPaymentsByStatus(Payment.PaymentStatus status) {
        return paymentRepository.findByStatus(status);
    }
    
    public Payment createPayment(Payment payment) {
        payment.setTransactionId(UUID.randomUUID().toString());
        return paymentRepository.save(payment);
    }
    
    public Payment processPayment(Long id) {
        Payment payment = paymentRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Payment not found"));
        
        payment.setStatus(Payment.PaymentStatus.PROCESSING);
        return paymentRepository.save(payment);
    }
    
    public Payment completePayment(Long id) {
        Payment payment = paymentRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Payment not found"));
        
        payment.setStatus(Payment.PaymentStatus.COMPLETED);
        payment.setPaymentDate(LocalDateTime.now());
        return paymentRepository.save(payment);
    }
    
    public Payment failPayment(Long id) {
        Payment payment = paymentRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Payment not found"));
        
        payment.setStatus(Payment.PaymentStatus.FAILED);
        return paymentRepository.save(payment);
    }
    
    public void deletePayment(Long id) {
        Payment payment = paymentRepository.findById(id)
                .orElseThrow(() -> new RuntimeException("Payment not found"));
        paymentRepository.delete(payment);
    }
}

File: frontend/src/services/paymentService.js
import axios from 'axios';

const API_URL = process.env.REACT_APP_API_URL || 'http://localhost:8080/api';

export const paymentService = {
    getAllPayments: async () => {
        const response = await axios.get(`${API_URL}/payments`);
        return response.data;
    },

    getPaymentById: async (id) => {
        const response = await axios.get(`${API_URL}/payments/${id}`);
        return response.data;
    },

    getPaymentByOrderId: async (orderId) => {
        const response = await axios.get(`${API_URL}/payments/order/${orderId}`);
        return response.data;
    },

    getPaymentsByStatus: async (status) => {
        const response = await axios.get(`${API_URL}/payments/status/${status}`);
        return response.data;
    },

    createPayment: async (payment) => {
        const response = await axios.post(`${API_URL}/payments`, payment);
        return response.data;
    },

    processPayment: async (id) => {
        const response = await axios.patch(`${API_URL}/payments/${id}/process`);
        return response.data;
    },

    completePayment: async (id) => {
        const response = await axios.patch(`${API_URL}/payments/${id}/complete`);
        return response.data;
    },

    failPayment: async (id) => {
        const response = await axios.patch(`${API_URL}/payments/${id}/fail`);
        return response.data;
    },

    deletePayment: async (id) => {
        await axios.delete(`${API_URL}/payments/${id}`);
    }
};
